app/src/main/java/org/resolvetosavelives/red/newentry/search/PatientRepository.kt

```diff
@@ -44,34 +44,64 @@ class PatientRepository @Inject constructor(private val database: AppDatabase) {
     })
   }
 
-  fun markOngoingEntryAsComplete(): Completable {
-    // TODO: Parse date and convert it to millis.
-    val dateConverter: (String) -> LocalDate? = { formattedDate -> null }
-
-    val patientUuid = UUID.randomUUID().toString()
+  fun saveOngoingEntryAsPatient(): Completable {
     val addressUuid = UUID.randomUUID().toString()
-    val phoneNumberUuid = UUID.randomUUID().toString()
 
-    val createdAtDateTime = Instant.now()
-    val updatedAtDateTime = createdAtDateTime
+    val addressSave = ongoingEntry()
+        .map { validEntry ->
+          with(validEntry) {
+            PatientAddress(
+                uuid = addressUuid,
+                colonyOrVillage = address!!.colonyOrVillage,
+                district = address.district,
+                state = address.state,
+                createdAt = Instant.now(),
+                updatedAt = Instant.now(),
+                syncPending = true)
+          }
+        }
+        .flatMapCompletable { address -> saveAddress(address) }
 
-    return ongoingEntry()
+    val patientSave = ongoingEntry()
         .map { entry ->
-          entry.apply {
+          with(entry) {
             Patient(
-                uuid = patientUuid,
-                addressUuid = addressUuid,
-                phoneNumberUuid = phoneNumberUuid,
+                uuid = UUID.randomUUID().toString(),
                 fullName = personalDetails!!.fullName,
                 gender = personalDetails.gender!!,
-                dateOfBirth = dateConverter(personalDetails.dateOfBirth),
-                ageWhenCreated = personalDetails.ageWhenCreated!!.toInt(),
                 status = PatientStatus.ACTIVE,
-                createdAt = createdAtDateTime,
-                updatedAt = updatedAtDateTime,
+
+                //todo: both, dateOfBirth and ageWhenCreated, should not be null at the same time
+                //todo: ageWhenCreated should be an int
+                dateOfBirth = dateConverter(personalDetails.dateOfBirth),
+                ageWhenCreated = personalDetails.ageWhenCreated?.toInt(),
+
+                addressUuid = addressUuid,
+                phoneNumberUuid = null,
+
+                createdAt = Instant.now(),
+                updatedAt = Instant.now(),
                 syncPending = true)
           }
         }
-        .flatMapCompletable { patient -> saveOngoingEntry(patient) }
+        .flatMapCompletable { patient -> savePatient(patient) }
+
+    return addressSave.andThen(patientSave)
+  }
+
+  private fun dateConverter(dateOfBirth: String): LocalDate? {
+    return LocalDate.parse(dateOfBirth, DateTimeFormatter.ofPattern("dd-MM-yyyy")) ?: null
```

Comment:
Oh nice you also added this

Reply:
This statement looks wrong…

`LocalDate.parse(dateOfBirth, DateTimeFormatter.ofPattern("dd-MM-yyyy")) ?: null`

It looks like we're saying if the parsed date is null then return null. I pull your branch on my computer and test this.

Reply:
Also, do you want to use `LocalDateRoomTypeConverter().toDateTime()` here?